src/test/java/seedu/guest/model/guest/RoomTest.java

```diff
@@ -0,0 +1,40 @@
+package seedu.guest.model.guest;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.guest.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class RoomTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Room(null));
+    }
+
+    @Test
+    public void constructor_invalidRoom_throwsIllegalArgumentException() {
+        String invalidRoom = "$50-23";
+        assertThrows(IllegalArgumentException.class, () -> new Room(invalidRoom));
+    }
+
+    @Test
+    public void isValidRoom() {
+        // null room
+        assertThrows(NullPointerException.class, () -> Room.isValidRoom(null));
+
+        // invalid room
+        assertFalse(Room.isValidRoom("")); // empty string
+        assertFalse(Room.isValidRoom(" ")); // spaces only
+        assertFalse(Room.isValidRoom("-50-23")); // not allowed to start with a hyphen
```

Comment:
Is it important to have a case where the room number is "06--09", where there is 2 hyphens?

Reply:
Sounds good!